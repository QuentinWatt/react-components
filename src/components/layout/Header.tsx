import { Link } from "react-router-dom";

const Header: React.FC = () => {
  return (
    <header className="border-b bg-white">
      <nav className="container mx-auto px-3">
        <div className="flex justify-between items-center h-16">
          <span className="font-bold text-3xl">Qcasts</span>
          <ul className="flex items-center">
            <li className="mr-2">
              <Link to="/">Home</Link>
            </li>
            <li className="mr-2">
              <Link to="/about">About</Link>
            </li>
          </ul>
        </div>
      </nav>
    </header>
  );
};

export default Header;
