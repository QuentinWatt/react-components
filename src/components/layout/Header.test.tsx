import { render, screen } from "@testing-library/react";
import Header from "./Header";

jest.mock("react-router-dom", () => ({
  Link: ({ to, children }: { to: string; children: React.ReactNode }) => (
    <a href={to}>{children}</a>
  ),
}));

describe("It has the correct header", () => {
  beforeEach(() => {
    render(<Header />);
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("has a copyright", () => {
    expect(screen.getByText("Qcasts")).toBeTruthy();
  });

  it("has the correct list items", () => {
    expect(screen.getByText("Home")).toBeTruthy();
    expect(screen.getByText("About")).toBeTruthy();
  });
});
