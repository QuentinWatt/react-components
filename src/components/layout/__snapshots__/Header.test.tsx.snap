// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It has the correct header matches the snapshot 1`] = `
<DocumentFragment>
  <header
    class="border-b bg-white"
  >
    <nav
      class="container mx-auto px-3"
    >
      <div
        class="flex justify-between items-center h-16"
      >
        <span
          class="font-bold text-3xl"
        >
          <a
            href="/"
          >
            Qcasts
          </a>
        </span>
        <ul
          class="flex items-center"
        >
          <li
            class="mr-2"
          >
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="mr-2"
          >
            <a
              href="/about"
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
